app/models/target.rb

```diff
@@ -27,4 +27,11 @@ class Target < ApplicationRecord
 
   validates :title, :longitude, :latitude, :radius, presence: true
   validates :radius, numericality: { only_integer: true }
+  validate :number_of_user_targets
+
+  private
+
+  def number_of_user_targets
```

Comment:
IMO this kind of validations are "business logic" that don't belong to the models layer. I'd create a service that handles targets creation and put this logic there

```ruby
class Api::V1:TargetsController
  def create
    Targets::CreateService.new(params).call
  end
end

module Targets
  class ExceededTargetsCount < StandardError; end

  class CreateService
    MAX_ALLOWED_TARGETS = 10

    def initialize(params)
      @params = params
    end

    def call
      raise ExceededTargetsCount unless user_targets_count < MAX_ALLOWED_TARGETS

      Target.create!(params)
    end
  end

  private

  def user_targets_count
    @user_targets_count ||= Target.where(user_id: params[:user_id]).count
  end
end
```

Reply:
@JulianPasquale great suggestion. Also, errors could go inside their own folder at `app/errors`